File: C++/cyclicsort.cpp
#include<bits/stdc++.h>
using namespace std;

int swap(int arr[],int first,int second){
    int temp=arr[first];
    arr[first]=arr[second];
    arr[second]=temp;
}




int main(){
    int n;
    cin>>n;

    int arr[n];

    for(int i=0;i<n;i++){
        cin>>arr[i];
    }
    int i=0;
    while(i<n){

        int correctindex=arr[i]-1;

        if(arr[i]!=arr[correctindex]){
            swap(arr,i,correctindex);
        } else{
            i++;
        }

    }

    for(int i=0;i<n;i++){
        cout<<arr[i]<<" ";
    }
}

File: Java/Cyclic_Sort.java
import java.util.Arrays;
import java.util.Scanner;

public class Cyclic_sort {
    public static void main(String[] args) {
        int n;
        Scanner s = new Scanner(System.in);
        System.out.print("Enter no. of elements you want in array:");
        n = s.nextInt();
        int[] arr = new int[n];
        System.out.println("Enter all the elements:");
        for (int i = 0; i < n; i++) {
            arr[i] = s.nextInt();
        }
        sort(arr);
        System.out.println(Arrays.toString(arr));
    }

    static void sort(int[] arr) {
        int i = 0;
        while (i < arr.length) {
            int correct = arr[i] - 1;
            if (arr[i] != arr[correct]) {
                swap(arr, i, correct);
            } else {
                i++;
            }
        }
    }

    static void swap(int[] arr, int first, int second) {
        int temp = arr[first];
        arr[first] = arr[second];
        arr[second] = temp;
    }
}
